src/psd_tools/api/layers.py

```diff
@@ -576,6 +688,22 @@ def __setitem__(self, key, value):
     def __delitem__(self, key):
         return self._layers.__delitem__(key)
 
+    # Use the layer structure methods instead, they make more comprehensive changes (with metadata updates)
```

Comment:
I'm not sure what these methods are intended for. Are you trying to emulate container types?
https://docs.python.org/3/reference/datamodel.html#emulating-container-types

Reply:
That was the goal yes, i first added them so that groups would be assimilated to python lists, and i use some of them in other structure modification methods, but i'm now not sure they're that relevant to have.

Reply:
Do you think i should move the add_layer(s) methods (and similar ones) to the corresponding container emulation ones or should i just remove them ?

Reply:
Container emulation is optional. I would suggest either of the following:

1. Implement the public method name, e.g., `append`, instead of the private name `_append`, following [the standard data structure](https://docs.python.org/3/tutorial/datastructures.html#more-on-lists)
2. Drop those private methods

Reply:
I went with the container emulation suggestion for the GroupMixin class, the only two list methods i didn't make are sort and reverse which i don't think are very relevant for layer groups. I also added the relevant tests for these methods.